src/libraries/Microsoft.PowerFx.Core/Public/Values/TableValue.cs

```diff
@@ -85,16 +87,40 @@ private static ErrorValue ArgumentOutOfRange(IRContext irContext)
         // Async because derived classes may back this with a network call. 
         public virtual async Task<DValue<RecordValue>> AppendAsync(RecordValue record)
         {
-            // fails by default
-            throw new System.NotImplementedException("It is not possible to append to a TableValue directly.");
+            throw ThrowNotImplementedException(MethodBase.GetCurrentMethod().Name);
+        }
+
+        public virtual async Task<DValue<RecordValue>> PatchCoreAsync(RecordValue originalRecord, RecordValue newRecord)
```

Comment:
>PatchCoreAsync [](http://example.com/codeflow?start=55&length=14)

This should be protected - only callable by PatchAsync.

Comment that parameters here are already resolved to logical names using this table's schema.